inference-engine/src/inference_engine/include/openvino/runtime/executable_network.hpp

```diff
@@ -0,0 +1,155 @@
+
+    /**
+     * @brief Get parameters of executeble graph function
+     *
+     * @return vector of paramter nodes
+     */
+    std::vector<std::shared_ptr<ngraph::op::v0::Parameter>> get_parameters() const;
+
+    /**
+     * @brief Get results of executeble graph function
+     *
+     * @return vector of result nodes
+     */
+    std::vector<std::shared_ptr<ngraph::op::v0::Result>> get_results() const;
+
+    /**
+     * @brief Creates an inference request object used to infer the network.
+     *
+     * The created request has allocated input and output blobs (that can be changed later).
+     *
+     * @return InferRequest object
+     */
+    InferRequest create_infer_request();
+
+    /**
+     * @brief Exports the current executable network.
+     *
+     * @see Core::ImportNetwork
+     *
+     * @param networkModel Network model output stream
+     */
+    void export_model(std::ostream& networkModel);
+
+    /**
+     * @brief Sets configuration for current executable network
+     *
+     * @param config Map of pairs: (config parameter name, config parameter value)
+     */
+    void set_config(const ie::ParamMap& config);
+
+    /** @brief Gets configuration for current executable network.
+     *
+     * The method is responsible to extract information
+     * which affects executable network execution. The list of supported configuration values can be extracted via
+     * ExecutableNetwork::get_metric with the SUPPORTED_CONFIG_KEYS key, but some of these keys cannot be changed
+     * dynamically, e.g. DEVICE_ID cannot changed if an executable network has already been compiled for particular
+     * device.
+     *
+     * @param name config key, can be found in ie_plugin_config.hpp
+     * @return Configuration parameter value
+     */
+    ie::Parameter get_config(const std::string& name) const;
+
+    /**
+     * @brief Gets general runtime metric for an executable network.
+     *
+     * It can be network name, actual device ID on
+     * which executable network is running or all other properties which cannot be changed dynamically.
+     *
+     * @param name metric name to request
+     * @return Metric parameter value
+     */
+    ie::Parameter get_metric(const std::string& name) const;
+
+    /**
+     * @brief Returns pointer to plugin-specific shared context
+     * on remote accelerator device that was used to create this ExecutableNetwork
+     * @return A context
+     */
+    std::shared_ptr<ie::RemoteContext> get_context() const;
+
+    /**
+     * @brief Checks if current ExecutableNetwork object is not initialized
+     * @return true if current ExecutableNetwork object is not initialized, false - otherwise
+     */
+    bool operator!() const noexcept;
+
+    /**
+     * @brief Checks if current ExecutableNetwork object is initialized
+     * @return true if current ExecutableNetwork object is initialized, false - otherwise
+     */
+    explicit operator bool() const noexcept;
```

Comment:
Can we use this new class in the ov::Core?

Reply:
Will be integrated in #7259